src/org/python/core/JavaProxyList.java

```diff
@@ -339,38 +371,17 @@ public PyObject __call__(PyObject key) {
     private static final PyBuiltinMethodNarrow listEqProxy = new ListMethod("__eq__", 1) {
         @Override
         public PyObject __call__(PyObject other) {
-            List jList = asList();
-            if (other.getType().isSubType(PyList.TYPE)) {
-                PyList oList = (PyList) other;
-                if (jList.size() != oList.size()) {
-                    return Py.False;
-                }
-                for (int i = 0; i < jList.size(); i++) {
-                    if (!Py.java2py(jList.get(i))._eq(oList.pyget(i)).__nonzero__()) {
-                        return Py.False;
-                    }
-                }
-                return Py.True;
-            } else {
-                Object oj = other.getJavaProxy();
-                if (oj instanceof List) {
-                    List oList = (List) oj;
-                    if (jList.size() != oList.size()) {
-                        return Py.False;
-                    }
-                    for (int i = 0; i < jList.size(); i++) {
-                        if (!Py.java2py(jList.get(i))._eq(
-                                Py.java2py(oList.get(i))).__nonzero__()) {
-                            return Py.False;
-                        }
-                    }
-                    return Py.True;
-                } else {
-                    return null;
```

Comment:
Any binary operation, and `__eq__` is not different (I think) should be allowed to say "I don't know, ask the other guy". When you do this in Python, you return `NotImplemented`, but down here in the works you return `null`.

Reply:
Agree, but here I think we do know were not equal?

Reply:
I think you only know it's a type you can't deal with.